pkg/virt-controller/watch/vmi_test.go

```diff
@@ -189,6 +189,12 @@ var _ = Describe("VirtualMachineInstance watcher", func() {
 		})
 	}
 
+	ignorePodPatches := func() {
```

Comment:
This is a hack and should not be required.

Reply:
If we don't do something similar, the test suite will fail because there are unexpected patches to pods. This is due to the fact that if a pod is not found, now it will create it in line:

```go
pod, err := c.clientset.CoreV1().Pods(vmi.GetNamespace()).Create(context.Background(), templatePod, v1.CreateOptions{})
		//this will fail since pod is nil
		if k8serrors.IsForbidden(err) && strings.Contains(err.Error(), "violates PodSecurity") {
			psaErr := fmt.Errorf("failed to create pod for vmi %s/%s, it needs a privileged namespace to run: %w", vmi.GetNamespace(), vmi.GetName(), err)
			c.recorder.Eventf(vmi, k8sv1.EventTypeWarning, FailedCreatePodReason, failedToRenderLaunchManifestErrFormat, psaErr)
			return &syncErrorImpl{psaErr, FailedCreatePodReason}
			return pod, &syncErrorImpl{psaErr, FailedCreatePodReason}
		}
```
and later on, in `syncPausedConditionToPod` function, we will patch the pod. Therefore, something similar is required, otherwise, the test will fail. Do you know another way around, less hacky?

Reply:
I don't understand how this fails. Is the Pod not there?

Reply:
No, in these tests where I've added the function `ignorePodPatched` the pod is not present before starting the `controller.Execute()` function. Therefore, it needs to be created. The function [sync](https://github.com/kubevirt/kubevirt/pull/10961/files#diff-16487fff595728eb5eeec01225a2203756ecad25d90e90d5feb13a9ff08df7ffR1203) creates it. Before the change, due to the fact that the pod is created, but it is not returned (therefore after `sync` [function](https://github.com/jcanocan/kubevirt/blob/56012c131325f2bb24714bf69ea40beb35624418/pkg/virt-controller/watch/vmi.go#L391) 
the variable `pod` is still `pod = nil`) this made that `updateStatus` [function](https://github.com/jcanocan/kubevirt/blob/56012c131325f2bb24714bf69ea40beb35624418/pkg/virt-controller/watch/vmi.go#L525) does not execute function `syncPausedConditionToPod` [function](https://github.com/jcanocan/kubevirt/blob/56012c131325f2bb24714bf69ea40beb35624418/pkg/virt-controller/watch/vmi.go#L1030). Now, function `syncPausedConditionToPod` is executed (because the pod != nil) and it triggers a patch in the pod. Therefore, it fails if `ignorePodPatches` is not added because the test cases don't expect pod patches.

Reply:
This function has been removed.